infra/templates/gitlab-bootstrap-iac-cicd/README.md

```diff
@@ -0,0 +1,199 @@
+# Initialize the Terraform environment
+terraform init
+
+# See what the deployment will do. No changes will be applied, but you can review the changes that will be applied in the next step
+terraform plan
+
+# Deploy the changes
+terraform apply
+```
+
+**Enable backend state**
+
+Enabling backend state will store the deployment state in Azure. This will allow others to run the deployment without you needing to worry about the state configuration.
+
+Start by uncommenting the contents of `backend.tf`, then run the following:
+
+```bash
+export ARM_ACCESS_KEY=$(terraform output backend-state-account-key)
+export ARM_ACCOUNT_NAME=$(terraform output backend-state-account-name)
+export ARM_CONTAINER_NAME=$(terraform output backend-state-bootstrap-container-name)
+
+# Initialize the deployment with the backend
+terraform init -backend-config "storage_account_name=${ARM_ACCOUNT_NAME}" -backend-config "container_name=${ARM_CONTAINER_NAME}"
+```
+
+You should see something along the lines of the following, to which you want to answer `yes`:
+
+```bash
+Do you want to copy existing state to the new backend?
+```
+
+If things work, you will see the following message and the state file should end up in Azure:
+
+```bash
+Successfully configured the backend "azurerm"! Terraform will automatically
+use this backend unless the backend configuration changes.
+```
+
+### Deploying the Infrastructure
+
+Now that Azure and GitLab have been configured to support the Terraform deployment, you will need to do the following to actually deploy the environment.
+
+**Trigger IAC Pipeline**
+
+You are now ready to kick off a deployment of the IAC pipeline! You can do this through the GitLab UI.
+
+### Rotate Service Principal Passwords
+
+If the need arrises to rotate the credentials for any of the generated service principals, the following command can be used to quickly rotate the credentials and also update all configuration in GitLab:
+
+```bash
+# configure environment (.envrc.template)
+
+az login
+az account set --subscription "<your subscription ID>"
+
+terraform init -backend-config "storage_account_name=${ARM_ACCOUNT_NAME}" -backend-config "container_name=${ARM_CONTAINER_NAME}"
+
+# `taint` all passwords - this triggers Terraform to regenerate these and update all dependent configuration
+terraform state list | grep random_password | xargs -L 1 terraform taint
+terraform plan
+
+# Note: this command might fail due to the rapid create/delete on Azure resources. If it fails, re-running it
+#       should solve the issue
+terraform apply
+```
+
+Done!
+
+
+### Adding a new environment
+
+Now that Azure and GitLab have been configured to deploy the `TEK` Server through Terraform, you can easily configure Azure and GitLab to support new stages by using the `environment` module.
+
+> **Note**: This will only set up Auzre and GitLab to support a new environment. The environment will need to be deployed using the infrastructure deployments project (not covered here).
```

Comment:
```suggestion
> **Note**: This will only set up Azure and GitLab to support a new environment. The environment will need to be deployed using the infrastructure deployments project (not covered here).
```